asf_public_discourse_home_decarbonisation/pipeline/data_processing_flows/flow_utils.py

```diff
@@ -0,0 +1,267 @@
+    """
+    categories = [
+        "green-ethical-moneysaving",
+        "lpg-heating-oil-solid-other-fuels",
+        "energy",
+        "is-this-quote-fair",
+    ]
+    all_mse_data = pd.DataFrame()
+    for cat in categories:
+        if (
+            cat != "energy"
+        ):  # this is a temporary fix, while we sort the collection for this category
+            aux = load_s3_data(
+                bucket_name=S3_BUCKET,
+                file_path=f"data/mse/outputs/mse_data_category_{cat}_{collection_date}.parquet",
+            )
+        else:
+            aux = load_s3_data(
+                bucket_name=S3_BUCKET,
+                file_path=f"data/mse/outputs/mse_data_category_energy.parquet",
+            )
+        all_mse_data = pd.concat([all_mse_data, aux])
+
+    return all_mse_data.drop_duplicates().reset_index(drop=True)
+
+
+def get_mse_data(category: str, collection_datetime: str) -> pd.DataFrame:
+    """
+    Get MSE data for a specific category and collection date.
+    Existing categories are: "green-ethical-moneysaving", "lpg-heating-oil-solid-other-fuels", "energy", "is-this-quote-fair"
+    Additionally, you can also use "sample" (to get the initial sample collected) or "all" (to get all the MSE data collected).
+
+    Args:
+        category (str): An MSE category, "sample" or "all"
+        collection_datetime (str): A date in the format "YYYY_MM_DD"
+
+    Returns:
+        pd.DataDrame: a dataframe with the MSE data
+    """
+    mse_categories = [
+        "green-ethical-moneysaving",
+        "lpg-heating-oil-solid-other-fuels",
+        "energy",
+        "is-this-quote-fair",
+    ]
+
+    accepted_categories = mse_categories + ["all"]
+    try:
+        # Check if the category is in the list if one of the accepted categories
+        category_index = accepted_categories.index(category)
+        if category in [
+            "green-ethical-moneysaving",
+            "lpg-heating-oil-solid-other-fuels",
+            "is-this-quote-fair",
+            "energy",
+        ]:
+            mse_data = get_mse_category_data(category, collection_datetime)
+        else:  # category == "all"
+            mse_data = get_all_mse_data(collection_datetime)
+        logger.info(f"Data for category '{category}' imported successfully from S3.")
+    except ValueError:
+        logger.error(f"{category} is not a valid category!")
+        sys.exit(-1)
+
+    return mse_data
+
+
+def remove_urls(text: str) -> str:
+    """
+    Removes URLs from text.
+    Args:
+        text (str): a string, tipically one or multiple sentences long
+    Returns:
+        str: text without URLs
+    """
+    pattern = r"http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\\(\\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+"
+    cleaned_text = re.sub(pattern, " ", text)
+    return cleaned_text
+
+
+def remove_text_after_patterns(text: str) -> str:
+    """
+    Removes pattern of the form "xxx writes: ".
```

Comment:
Should it be "xxx wrote" or am I misunderstanding the defined variable of result?